tests/Fixer/Phpdoc/PhpdocTypesFixerTest.php

```diff
@@ -233,4 +233,54 @@ public function testWrongConfig()
 
         $this->fixer->configure(['groups' => ['__TEST__']]);
     }
+
+    public function testRemoveSpacesBeforeAndAfterPipe()
+    {
+        $expected = <<<'EOF'
+<?php
+    /**
+     * @param self|array|Foo $bar
+     *
+     * @return int|float|callback
+     */
+
+EOF;
+
+        $input = <<<'EOF'
+<?php
+    /**
+     * @param self | array | Foo $bar
```

Comment:
```
Description of phpdoc_types rule.
The correct case must be used for standard PHP types in PHPDoc.
```

Removing extra spaces has nothing to do with this fixer.
Your feature looks nice, but you can't just pick random fixer and add new logic there.
Make new rule (no spaces between types) as new fixer.